pkg/extensions/tbtc/tbtc.go

```diff
@@ -761,23 +803,65 @@ func (t *tbtc) watchKeepClosed(
 }
 
 func (t *tbtc) shouldMonitorDeposit(depositAddress string) bool {
-	keepAddress, err := t.chain.KeepAddress(depositAddress)
+	t.monitoredDepositsCache.Sweep()
+	t.notMonitoredDepositsCache.Sweep()
+
+	if t.monitoredDepositsCache.Has(depositAddress) {
+		return true
+	}
+
+	if t.notMonitoredDepositsCache.Has(depositAddress) {
+		return false
+	}
+
+	signerIndex, err := t.getSignerIndex(depositAddress)
 	if err != nil {
+		logger.Errorf(
+			"could not get signer index for deposit [%v]: [%v]",
+			depositAddress,
+			err,
+		)
+		return false // return false but don't cache the result in case of error
```

Comment:
This is a good decision but I'd add some more context to the error we log about what are the implications of it. My proposition:
```
could not check if deposit [%v] should be monitored: failed to get signer index: [%v]`
```